homeassistant/components/modbus/fan.py

```diff
@@ -0,0 +1,41 @@
+"""Support for Modbus fans."""
+import logging
+from typing import Optional
+
+from homeassistant.components.fan import FanEntity
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import CONF_FANS, MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusSwitch
```

Comment:
Please refactor to not depend on another platform.

Reply:
Do you mean the `Switch` platform? The idea was to build `Light` on top of `Switch` because it does the same functionality. `Light` class can be then further extended to add other feature.

If I can't depend on `Switch`, basically I will need to duplicate (copy) the `Switch` code. It's not a problem, I just want to clarify my approach with you.

Reply:
@vzahradnik I would do that for now, we need your 2 new platforms. btw. did you also update the documentation?

Reply:
Yes, the documentation is updated. So I will duplicate the code for now, and later we'll refactor.